dbms/src/Dictionaries/DictionaryBlockInputStream.h

```diff
@@ -0,0 +1,290 @@
+                column = getColumnFromAttribute<TYPE, Getter<TYPE>>( \
+                &DictionaryType::get##TYPE, ids, key_columns_palin_ptrs, data_types, attribute, dictionary)
+            switch (attribute.underlying_type)
+            {
+                case AttributeUnderlyingType::UInt8: GET_COLUMN_FORM_ATTRIBUTE(UInt8); break;
+                case AttributeUnderlyingType::UInt16: GET_COLUMN_FORM_ATTRIBUTE(UInt16); break;
+                case AttributeUnderlyingType::UInt32: GET_COLUMN_FORM_ATTRIBUTE(UInt32); break;
+                case AttributeUnderlyingType::UInt64: GET_COLUMN_FORM_ATTRIBUTE(UInt64); break;
+                case AttributeUnderlyingType::Int8: GET_COLUMN_FORM_ATTRIBUTE(Int8); break;
+                case AttributeUnderlyingType::Int16: GET_COLUMN_FORM_ATTRIBUTE(Int16); break;
+                case AttributeUnderlyingType::Int32: GET_COLUMN_FORM_ATTRIBUTE(Int32); break;
+                case AttributeUnderlyingType::Int64: GET_COLUMN_FORM_ATTRIBUTE(Int64); break;
+                case AttributeUnderlyingType::Float32: GET_COLUMN_FORM_ATTRIBUTE(Float32); break;
+                case AttributeUnderlyingType::Float64: GET_COLUMN_FORM_ATTRIBUTE(Float64); break;
+                case AttributeUnderlyingType::String: 
+                {
+                    column = getColumnFromStringAttribute<StringGetter>(
+                        &DictionaryType::getString, ids, key_columns_palin_ptrs, data_types, attribute, dictionary);
+                    break;
+                }
+            }
+
+            columns.emplace_back(column, attribute.type, attribute.name);
+        }
+    }
+    block = Block(columns);
+}
+
+template <class DictionaryType, class Key>
+template <class AttributeType, class Getter>
+ColumnPtr DictionaryBlockInputStream<DictionaryType, Key>::getColumnFromAttribute(
+    Getter getter, const PaddedPODArray<Key> & ids,
+    const ConstColumnPlainPtrs & keys, const DataTypes & data_types,
+    const DictionaryAttribute & attribute, const DictionaryType & dictionary)
+{
+    auto size = ids.size();
+    if (!keys.empty())
+        size = keys.front()->size();
+    auto column_vector = std::make_unique<ColumnVector<AttributeType>>(size);
+    callGetter(getter, ids, keys, data_types, column_vector->getData(), attribute, dictionary);
+    return ColumnPtr(std::move(column_vector));
+}
+
+template <class DictionaryType, class Key>
+template <class Getter>
+ColumnPtr DictionaryBlockInputStream<DictionaryType, Key>::getColumnFromStringAttribute(
+    Getter getter, const PaddedPODArray<Key> & ids,
+    const ConstColumnPlainPtrs & keys, const DataTypes & data_types,
+    const DictionaryAttribute& attribute, const DictionaryType& dictionary)
+{
+    auto column_string = std::make_unique<ColumnString>();
+    auto ptr = column_string.get();
+    callGetter(getter, ids, keys, data_types, ptr, attribute, dictionary);
+    return ColumnPtr(std::move(column_string));
```

Comment:
No need for std::move, you could create shared_ptr instead.
No need for explicit cast to ColumnPtr.